web/core/src/jsTest/kotlin/CSSUnitApiTests.kt

```diff
@@ -306,6 +273,162 @@ class CSSUnitApiTests {
+        assertEquals("calc(4ex + 4em)", typedResultRel.toString())
+
+        val typedResultLength: CSSNumericValue<CSSUnitLength> = 4.pt + 4.em
+        assertEquals("calc(4pt + 4em)", typedResultLength.toString())
+
+        val typedResultLengthOrPercentage: CSSNumericValue<CSSUnitLengthOrPercentage> = 4.percent + 4.px
+        assertEquals("calc(4% + 4px)", typedResultLengthOrPercentage.toString())
+
+        val typedResultAngle: CSSNumericValue<CSSUnitAngle> = 4.deg + 4.grad
+        assertEquals("calc(4deg + 4grad)", typedResultAngle.toString())
+
+        val typedResultTime: CSSNumericValue<CSSUnitTime> = 4.s + 4.ms
+        assertEquals("calc(4s + 4ms)", typedResultTime.toString())
+
+        val typedResultFrequency: CSSNumericValue<CSSUnitFrequency> = 4.Hz + 4.kHz
+        assertEquals("calc(4Hz + 4kHz)", typedResultFrequency.toString())
+
+        val typedResultResolution: CSSNumericValue<CSSUnitResolution> = 4.dpi + 4.dppx
+        assertEquals("calc(4dpi + 4dppx)", typedResultResolution.toString())
+    }
+
+    @Test
+    fun calcSubstract() {
+        val typedResultAbs: CSSNumericValue<CSSUnitAbs> = 4.pt - 4.px
+        assertEquals("calc(4pt - 4px)", typedResultAbs.toString())
+
+        val typedResultRel: CSSNumericValue<CSSUnitRel> = 4.ex - 4.em
+        assertEquals("calc(4ex - 4em)", typedResultRel.toString())
+
+        val typedResultLength: CSSNumericValue<CSSUnitLength> = 4.pt - 4.em
+        assertEquals("calc(4pt - 4em)", typedResultLength.toString())
+
+        val typedResultLengthOrPercentage: CSSNumericValue<CSSUnitLengthOrPercentage> = 4.percent - 4.px
+        assertEquals("calc(4% - 4px)", typedResultLengthOrPercentage.toString())
+
+        val typedResultAngle: CSSNumericValue<CSSUnitAngle> = 4.deg - 4.grad
+        assertEquals("calc(4deg - 4grad)", typedResultAngle.toString())
+
+        val typedResultTime: CSSNumericValue<CSSUnitTime> = 4.s - 4.ms
+        assertEquals("calc(4s - 4ms)", typedResultTime.toString())
+
+        val typedResultFrequency: CSSNumericValue<CSSUnitFrequency> = 4.Hz - 4.kHz
+        assertEquals("calc(4Hz - 4kHz)", typedResultFrequency.toString())
+
+        val typedResultResolution: CSSNumericValue<CSSUnitResolution> = 4.dpi - 4.dppx
+        assertEquals("calc(4dpi - 4dppx)", typedResultResolution.toString())
+    }
+
+    @Test
+    fun calcAssociative() {
+        val typedResultLengthLeft: CSSNumericValue<CSSUnitLength> = 4.pt - 4.px + 4.em
+        assertEquals("calc((4pt - 4px) + 4em)", typedResultLengthLeft.toString())
+
+        val typedResultLengthRight: CSSNumericValue<CSSUnitLength> = 4.pt - (4.px + 4.em)
+        assertEquals("calc(4pt - (4px + 4em))", typedResultLengthRight.toString())
+    }
+
+    @Test
+    fun calcVaraiables() {
+        val variables = object : CSSVariables {
+            val pxVar by variable<CSSSizeValue<CSSUnit.px>>()
+        }
+        val typedResultLength: CSSNumericValue<CSSUnitLength> = 4.pt + variables.pxVar.value()
```

Comment:
What do you think about adding more operators overloads to make it possible to write like this:

```
4.pt + variables.pxVar
```

In this case it needs an overload:
```
operator fun <T: CSSUnit> CSSNumericValue<out T>.plus(b: CSSStyleVariable<CSSNumericValue<out T>>): CSSCalcValue<T> = CSSCalcValue(CSSPlus(this, b.value()))
```

I would prefer to not write ".value()" if I don't need a fallback.

Reply:
we can do that, but it will add some ambiguity - two ways to do same things. I'll think about different cases.